modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorMatchedSlotSubFetchPhase.java

```diff
@@ -96,7 +97,30 @@ public void process(HitContext hitContext) throws IOException {
 
                     IntStream slots = convertTopDocsToSlots(topDocs, pc.rootDocsBySlot);
                     // _percolator_document_slot fields are document fields and should be under "fields" section in a hit
-                    hitContext.hit().setDocumentField(fieldName, new DocumentField(fieldName, slots.boxed().collect(Collectors.toList())));
+                    List<Object> docSlots = slots.boxed().collect(Collectors.toList());
+                    hitContext.hit().setDocumentField(fieldName, new DocumentField(fieldName, docSlots));
+
+                    // Add info what sub-queries of percolator query matched this each percolated document
+                    if (fetchContext.getSearchExecutionContext().hasNamedQueries()) {
+                        List<LeafReaderContext> leafContexts = percolatorIndexSearcher.getLeafContexts();
```

Comment:
Why can't we use the `ctx` that is set via 
```
public void setNextReader(LeafReaderContext readerContext) {
                this.ctx = readerContext;
            }
```
?

Reply:
These are two different readerContext.

The one you suggested in `setNextReader` is for an index storing percolator queries.

But here, we need readerContext for an in-memory index storing percolating documents.